### PR TITLE
Added back previously fixed bugs

### PR DESCRIPTION
**Pull request by ctl43**

### CODE REVIEW

File: R/multiReadAlign.R [hunk 1]
```diff
@@ -5,7 +5,7 @@
 #' @importFrom methods is new
 #' @importFrom S4Vectors metadata
 #' @importClassesFrom S4Vectors DataFrame
-multiReadAlign <- function(reads, groups, max.error=0.1, keep.masked=FALSE, ..., BPPARAM=SerialParam())
+multiReadAlign <- function(reads, groups, max.error=1, keep.masked=FALSE, ..., BPPARAM=SerialParam())
```
Comment (LTLA) [inline R/multiReadAlign.R]: Set to `NA` to turn off.

File: R/expectedDist.R [hunk 2]
```diff
@@ -44,7 +44,7 @@ expectedDist <- function(align.stats, position=NULL, offset=NULL, number=100, ge
     combined <- DNAStringSet(paste0(front.seq, back.seq))
     if (is(front.seq, "QualityScaledDNAStringSet")) {
         combined.qual <- paste0(quality(front.seq), quality(back.seq))
-        combined <- QualityScaledDNAStringSet(combined, combined.qual)
+        combined <- QualityScaledDNAStringSet(combined, PhredQuality(combined.qual))
```
Comment (LTLA) [inline R/expectedDist.R]: There are many different qualities (e.g., Solexa, Illumina) - make sure you use the same quality type as the input.

File: R/sam2ranges.R [hunk 3]
```diff
@@ -13,22 +14,23 @@ sam2ranges <- function(sam, minq = 10, restricted = NULL)
...
     repeat {
         curline <- readLines(curfile, n=1)
         if (!grepl("^@", curline)) {
             break
         }
         if (grepl("^@SQ", curline)) {
-            sections <- strsplit(curline, "\t")[[1]]
-            curname <- sub("^SN:", "", sections[[2]])
-            curlen <- sub("^LN:", "", sections[[3]])
-            collected[[curname]] <- as.integer(curlen)
```
Comment (LTLA) [inline R/sam2ranges.R:28]: WTF why did you remove this?

Comment (ctl43) [reply R/sam2ranges.R:28]: Look at what I added later.
I collect all the seqnames first and then split them in one shot instead of splitting them within the loop. (just for efficiency)

File: R/expectedDist.R (hunk 2 above)
Comment (ctl43) [reply R/expectedDist.R]: Fixed

File: R/multiReadAlign.R (hunk 1 above)
Comment (ctl43) [reply R/multiReadAlign.R]: Fixed
